### PR TITLE
ISBD-266:

### PR DESCRIPTION
**Pull request by Murshadhaque**
A59 Form validation implementation

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by rahulsanjava19 — APPROVED** [review]
LGTM

File: src/functionalTest/resources/requests/bulk-scan-a59-transform-input.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+  "ocr_data_fields":[
+    {
+      "name":"applicant1_firstName",
```
Comment (yogendraU) [inline src/functionalTest/resources/requests/bulk-scan-a59-transform-input.json:4]: please check the convention what we have in master and update accordingly.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA59Service.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.bulkscan.constants.BulkScanConstants.BULK_SCAN_CASE_REFERENCE;
+import static uk.gov.hmcts.reform.bulkscan.constants.BulkScanConstants.CASE_TYPE_ID;
+import static uk.gov.hmcts.reform.bulkscan.constants.BulkScanConstants.EVENT_ID;
+
+
+@Service
+public class BulkScanA59Service implements BulkScanService {
```
Comment (yogendraU) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA59Service.java:28]: Is this class in sycn with the framework in master code

File: src/main/resources/application-A59-transform.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      transform:
+        caseTypeofApplication:
+          A59:
+            caseFields:
+              CASE_TYPE_ID: A59
+              EVENT_ID: create-case-adop-bs
+            caseDataFields:
+              applicant:
+                - value:
+                    applicant1FirstName: applicant1_firstName
```
Comment (yogendraU) [inline src/main/resources/application-A59-transform.yaml:13]: Please update this

**Review by yogendraU — CHANGES_REQUESTED** [review]

File: src/main/resources/application-A59-transform.yaml (hunk 3 above)
Comment (rahulsanjava19) [reply src/main/resources/application-A59-transform.yaml:13]: good catch :-)

File: src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA59Service.java (hunk 2 above)
Comment (Murshadhaque) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA59Service.java:29]: Done

File: src/functionalTest/resources/requests/bulk-scan-a59-transform-input.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{
+  "ocr_data_fields":[
+    {
+      "name":"applicant_firstName",
+      "value":"applicant1_firstName"
+    },
+    {
+      "name":"applicant_lastName",
+      "value":"applicant1_lastName"
```
Comment (yogendraU) [inline src/functionalTest/resources/requests/bulk-scan-a59-transform-input.json:9]: Why is is referring to applicant1_lastName in some places and directly to value for example for telephone number

File: src/main/resources/application-A59-form.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      form-validation:
+        caseTypeofApplication:
+          A59:
+            mandatoryFields:
+              - applicant1_firstName
+              - applicant1_lastName
+              - applicant1_address
+              - applicant1_telephoneNumber
+              - applicant2_firstName
+              - applicant2_lastName
```
Comment (yogendraU) [inline src/main/resources/application-A59-form.yaml:13]: Applicant 2 fields are also mandatory?

File: src/main/resources/application.yaml [hunk 6]
```diff
@@ -32,32 +32,12 @@ spring:
       - a60-form
       - c100-transform
       - a58-form
+      - A59-form
```
Comment (yogendraU) [inline src/main/resources/application.yaml:35]: Can we follow what is existing naming standards

**Review by yogendraU — CHANGES_REQUESTED** [review]

File: src/main/resources/application-A59-form.yaml (hunk 5 above)
Comment (Murshadhaque) [reply src/main/resources/application-A59-form.yaml:13]: yes

File: src/functionalTest/resources/requests/bulk-scan-a59-transform-input.json (hunk 4 above)
Comment (Murshadhaque) [reply src/functionalTest/resources/requests/bulk-scan-a59-transform-input.json:9]: we can put any value, value could be anything

File: src/main/resources/application.yaml (hunk 6 above)
Comment (Murshadhaque) [reply src/main/resources/application.yaml]: Done

Comment (Murshadhaque) [reply src/main/resources/application.yaml]: Corrected file names

File: src/functionalTest/resources/responses/bulk-scan-a59-validation-output.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    "items":[
+
+    ]
+  },
+  "errors":{
+    "items":[
+
+    ]
+  }
+}
```
Comment (hmcts-kodatala) [inline src/functionalTest/resources/responses/bulk-scan-a59-validation-output.json:13]: pls check this json is correct, status should be success([bulk-scan-a59-validation-output.json] for functional tests

File: src/test/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA58ServiceTest.java [hunk 8]
```diff
@@ -87,29 +84,16 @@ void testA58OptionalFieldsWarningsWhileDoingValidation() {
...
-                                                .build())
-                                        .build()))
+                        .scannedDocuments(TestDataUtil.getScannedDocumentsList())
                         .ocrdatafields(TestDataUtil.getA60OrC63orA58Data()).build());
 
         JSONAssert.assertEquals(readFileFrom(A58_STEP_PARENT_TRANSFORM_RESPONSE_PATH),
                 mapper.writeValueAsString(bulkScanTransformationResponse), true);
 
     }
-
```
Comment (hmcts-kodatala) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA58ServiceTest.java:114]: pls rollback from master, as it not related your changes

**Review by hmcts-kodatala — APPROVED** [review]
Ok

**Review by saravanakumar-palanichamy — APPROVED** [review]

**Pull request merged**